kernel/arch/riscv/debug.c

```diff
@@ -92,7 +92,7 @@ stackwalk_riscv(void **frames,
 void dump_stacktrace(void *ebp, pdir_t *pdir)
 {
    void *frames[32] = {0};
-   size_t c = stackwalk32(frames, ARRAY_SIZE(frames), ebp, pdir);
+   size_t c = stackwalk_riscv(frames, ARRAY_SIZE(frames), ebp, pdir);
```

Comment:
@aenrbes Oops, that means that the previous change broke the RISCV64 build :-(
That's why Tilck has a matrix of CI builds and tests for multiple configurations with multiple compilers. Unfortunately, we don't have one for the RISCV build yet and it will take me some time for me to do that as it's not that simple. The problem with breaking builds even for a single commit is that it prevents `git bisect` from working properly: if the build is broken at that commit, we cannot run a test and check if the regression was introduced at that point or not.

Reply:
I'm really sorry, this is really a mistake of mine, is there anything I can do to remedy it now?

Reply:
@aenrbes I could do a force-push to fix the issue if that won't be a problem for you. I avoid doing that, but if the commit is very recent, it's highly unlikely to upset many people because of that, at least for a project of this scale. For more popular projects that would never be an option.

Reply:
I don't have a problem, then please revoke the commit "[riscv64] Implement exception handling and panic", I will submit a new PR immediately

Reply:
@aenrbes I wanted to do it myself a very quick fix + `commit --amend` + `push --force` on the previous change, but waited for you to confirm first. There was no need to prepare a PR on the previous commit for a force push. Anyway, it got late in my time zone (Pacific Time) so I disconnected and this morning was already too late for the force-push.

Anyway, no big deal, the RISCV build is still experimental. Just, as the port is getting mature and things start working, we need to be careful to not break it.

Btw, on the full RISCV change did you run `build/st/run_all_tests` ? That's the script runner that runs all the selftests and also the devshell tests.

Reply:
@aenrbes Btw, can you rebase this PR on master?

Reply:
I tried to run `build/st/run_all_tests`, but found that it did not work properly with riscv64, and it may would take a pretty big fix to make `run_all_tests` work, so I manually ran all devshell and selftest commands. In addition, I tested the stability by executing `runall` continuously.

Reply:
It is worth mentioning that in the `selftest time` test, time drift will be significantly larger than i386,but this is expected and is not a bug; `fork_perf` is significantly slower than the i386, possibly due to three-level page tables, etc

Reply:
That's totally OK, thanks for the clarification. The time drift is not a big deal, nor the slower fork_perf. We can optimize that by implement CoW for the page tables. Currently, CoW during fork() is applied only for the regular pages, while all the page tables are just copied.

I'll take a look at that test runner once the full RISCV64 port is completed. Given that on RISCV64 the Tilck runs with the serial port, I guess there won't be a problem to make that runner work, because that's what's used there as well. Just, on i386, for performance and convenience reasons, I ask QEMU to load Tilck instead of Tilck's bootloader. That's also what the ` run_multiboot_qemu` script does (it uses QEMU's `-kernel` option). 

Therefore, for RISCV64 the kernel command line needs to be passed to the U-BOOT bootloader in a different way, maybe by writing to that special config file, I don't know. In the worst case, the test runner can write to serial port any command it wants and run the tests that way.